pkg/apis/openfaas/v1/types.go

```diff
@@ -111,6 +111,26 @@ type ProfileSpec struct {
 	//
 	// +optional
 	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+
+	// If specified, the pod's DNS policy.
+	//
+	// Valid values are 'ClusterFirstWithHostNet', 'ClusterFirst', 'Default' or 'None'.
+	// DNS parameters given in DNSConfig will be merged with the policy selected with DNSPolicy.
+	// To have DNS options set along with hostNetwork, you have to specify DNS policy
+	// explicitly to 'ClusterFirstWithHostNet'.
+	//
+	// copied to the Pod DNSPolicy, this will replace any existing value or previously
+	// applied Profile.
+	//
```

Comment:
It would be good to document and maybe link to the k8s documentation that "Default" is not actually the default value, it is actually `ClusterFirst` which you code properly implements, but I had to double check the docs to be sure. It will be easier to maintain if we add a note here

> Note: "Default" is not the default DNS policy. If dnsPolicy is not explicitly specified, then "ClusterFirst" is used.

from https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy